### PR TITLE
Tweaks for production

### PR DESCRIPTION
**Pull request by dfulu**
# Pull Request

## Description

- Add nan filling
- Make the target GSP location added to batch independent of adding the GSP data
- Eager-load the GSP data - not necessarily for production but I think we should have this in memory since its small enough

## How Has This Been Tested?

- Has been run with PVNet app

## Checklist:

- [x] My code follows [OCF's coding style guidelines](https://github.com/openclimatefix/.github/blob/main/coding_style.md)
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: ocf_data_sampler/torch_datasets/pvnet_uk_regional.py [hunk 1]
```diff
@@ -406,9 +425,18 @@ def process_and_combine_datasets(
...
-
-    # Combine all the modalities
+    
+    # Add coordinate data
+    # TODO: Do we need all of these?
+    numpy_modalities.append({
+        BatchKey.gsp_id: location.id,
+        BatchKey.gsp_x_osgb: location.x,
+        BatchKey.gsp_y_osgb: location.y,
+    })
```
Comment (peterdudfield) [inline ocf_data_sampler/torch_datasets/pvnet_uk_regional.py:435]: Id be tempted to keep them, nice to have when exploring a batch

File: ocf_data_sampler/torch_datasets/pvnet_uk_regional.py [hunk 2]
```diff
@@ -423,22 +451,6 @@ def compute(xarray_dict: dict) -> dict:
     return xarray_dict
 
 
-def get_locations(ga_gsp: xr.DataArray) -> list[Location]:
```
Comment (peterdudfield) [inline ocf_data_sampler/torch_datasets/pvnet_uk_regional.py:426]: is this just not needed now? i cant remember what is was needed for?

**Review by peterdudfield — APPROVED** [review]

File: ocf_data_sampler/torch_datasets/pvnet_uk_regional.py (hunk 2 above)
Comment (dfulu) [reply ocf_data_sampler/torch_datasets/pvnet_uk_regional.py:426]: No its not used anymore. `get_gsp_locations()` is used instead. I just forgot to delete this when that new function was added

File: ocf_data_sampler/torch_datasets/pvnet_uk_regional.py [hunk 3]
```diff
@@ -344,6 +344,24 @@ def slice_datasets_by_time(
     return sliced_datasets_dict
 
 
+def fill_nans_in_arrays(batch: NumpyBatch) -> NumpyBatch:
```
[the remark this answers is not in the document]
Comment (dfulu) [reply ocf_data_sampler/torch_datasets/pvnet_uk_regional.py:347]: I'd leave it in here for now. I think things like this should stay in the dataset file until they are used in multiple datasets and thus prove themselves as being generally useful

**Pull request merged**